syntax = "proto3";

package currency;

service CurrencyService {
  rpc Convert (ConvertRequest) returns (ConvertResponse);
}

message ConvertRequest {
  string from_currency = 1;
  string to_currency = 2;
  double amount = 3;
}

message ConvertResponse {
  double converted_amount = 1;
  double rate = 2;
  string message = 3;
}
